extensions/amp-experiment/1.0/mutation/attribute-mutation-default-url.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assertAttributeMutationFormat} from './mutation';
+import {assertHttpsUrl} from '../../../../src/url';
+
+/**
+ * Mutation for attribute (url) mutations, for both
+ * the 'href' and 'src' attributes,
+ * on unspecifed elements.
+ *
+ * @implements {./mutation.Mutation}
+ */
+export class AttributeMutationDefaultUrl {
+  /**
+   * @param {!JsonObject} mutationRecord
+   * @param {!Array<Element>} elements
+   */
+  constructor(mutationRecord, elements) {
+    this.mutationRecord = mutationRecord;
+    this.elements = elements;
+    assertAttributeMutationFormat(mutationRecord);
+  }
+
+  /** @override */
+  validate() {
+    assertHttpsUrl(
+      this.mutationRecord['value'],
+      this.mutationRecord['target'],
```

Comment:
mutationRecord['target'] is only a selector string. We also don't want to pass in the element because there could be multiple element selected. 
Looked at other use case of `assertHttpsUrl`. I think we can instead pass in a predefined msg. For example `amp-experiment url mutation`?

Reply:
Oh so I think this would be correct.

Looking at the function definition: https://github.com/ampproject/amphtml/blob/master/src/url.js#L308

The first param is the one that is used for assertion. The rest are used just for logging 😄 Thus, I think have the second param be the selector in the mutation, is a better developer experience. As I would personally want to know which mutation is causing the issue, rather than try to figure out which mutation is selecting my element.

Does that make sense? Happy to chat in person about this 😄

Reply:
I think that makes sense. But to me, only having the `mutationRecord['target']` info can be difficult to understand. do you think it's better if we can pass in the error message along with the mutation information?